### PR TITLE
.update openssl to 1.1.1i and added aarch64 support, fixed static vs.…

### PR DESCRIPTION
**Pull request by daxcore**
fix for #37

### CODE REVIEW

File: recipes/libs/openssl.yaml [hunk 1]
```diff
@@ -13,24 +13,68 @@ depends:
...
-    digestSHA256: 1e3a91bc1f9dfce01af26026f856e064eab4c8ee0a8f457b5ae30b40b8b711f2
+    digestSHA256: e8be6a35fe41d10603c3cc635e93289ed00bf34b79671a3a4de64fcee00d5242
     extract: False
 
 buildTools: [target-toolchain]
-buildVars: [PKG_VERSION, CC, AR, RANLIB, ARCH, AUTOCONF_BUILD, AUTOCONF_HOST]
+buildVars: [PKG_VERSION, CC, AR, RANLIB, ARCH, AUTOCONF_BUILD, AUTOCONF_HOST, BASEMENT_LIBS]
 buildScript: |
     export TARGETMACH=${AUTOCONF_HOST}
-    export LDFLAGS="-Wl,-rpath-link=${BOB_DEP_PATHS[libs::zlib-dev]}/usr/lib"
```
Comment (daxcore) [inline recipes/libs/openssl.yaml:23]: the rpath stuff isn't a good idea for being relocatable, is it?

Comment (jkloetzke) [reply recipes/libs/openssl.yaml:23]: The rpath-link does not end up in the binary. It's there to find transitive shared library dependencies. The cpackage class normally takes care of this and could indeed be used here. Did you test if it still builds in all cases after removing this line?

https://stackoverflow.com/questions/24598047/why-does-ld-need-rpath-link-when-linking-an-executable-against-a-so-that-needs

File: recipes/libs/openssl.yaml [hunk 2]
```diff
@@ -40,6 +84,10 @@ buildScript: |
 
     popd
 
+    if [[ "$SHARED_ONLY" == "true" ]] ; then
+        find install -iname "*.a" -exec rm -rf {} \;
+    fi
+
```
Comment (jkloetzke) [inline recipes/libs/openssl.yaml:90]: Why is this necessary? The install class takes care of this already. Static libraries are only kept in -dev packages.

Comment (daxcore) [reply recipes/libs/openssl.yaml:90]: for both, openssl and zlib, it isn't possible to just build shared libraries. static libraries will be built everytime, and shared libs are optional and additional by flag. but than "both" and "shared" are the same result.

btw. if i choose "both", which one will be used implicitly?

File: recipes/libs/openssl.yaml (hunk 1 above)
Comment (daxcore) [reply recipes/libs/openssl.yaml:23]: Not removed anymore, just added to the configure directly.

**Comment by jkloetzke** [comment]
Could you please structure this as two commits: one for zlib and one for OpenSSL? Please stick to the convention used in the other commits for the commit message.

Have you re-built the sandbox with these changes?

**Comment by daxcore** [comment]
> Could you please structure this as two commits: one for zlib and one for OpenSSL? Please stick to the convention used in the other commits for the commit message.

Okay, of course.

> Have you re-built the sandbox with these changes?

I will check this!

**Comment by daxcore** [comment]
done!?
